openedx/core/djangoapps/call_stack_manager/core.py

```diff
@@ -23,122 +27,193 @@
 
 log = logging.getLogger(__name__)
 
-# list of regular expressions acting as filters
+# List of regular expressions acting as filters
 REGULAR_EXPS = [re.compile(x) for x in ['^.*python2.7.*$', '^.*<exec_function>.*$', '^.*exec_code_object.*$',
                                         '^.*edxapp/src.*$', '^.*call_stack_manager.*$']]
-# Variable which decides whether to track calls in the function or not. Do it by default.
-TRACK_FLAG = True
 
-# List keeping track of Model classes not be tracked for special cases
-# usually cases where we know that the function is calling Model classes.
+# List keeping track of entities not to be tracked
 HALT_TRACKING = []
 
-# Module Level variables
-# dictionary which stores call stacks.
-# { "ModelClasses" : [ListOfFrames]}
+#  Dictionary which stores call logs
+# {'EntityName' : SetOf(ListOfFrames)}
 # Frames - ('FilePath','LineNumber','Context')
-# ex. {"<class 'courseware.models.StudentModule'>" : [[(file,line number,context),(---,---,---)],
-#                                                    [(file,line number,context),(---,---,---)]]}
+#  {"<class 'courseware.models.StudentModule'>" : [[(file, line number, function name, context),(---,---,---)],
+#                                                 [(file, line number, function name, context),(---,---,---)]]}
 STACK_BOOK = collections.defaultdict(list)
 
 
-def capture_call_stack(current_model):
-    """ logs customised call stacks in global dictionary `STACK_BOOK`, and logs it.
+def capture_call_stack(entity_name):
+    """ Logs customised call stacks in global dictionary STACK_BOOK and logs it.
 
-    Args:
-        current_model - Name of the model class
+    Arguments:
+        entity_name - entity
     """
-    # holds temporary callstack
-    # frame[0][6:-1] -> File name along with path
-    # frame[1][6:] -> Line Number
-    # frame[2][3:] -> Context
-    temp_call_stack = [(frame[0][6:-1],
-                        frame[1][6:],
-                        frame[2][3:])
-                       for frame in [stack.replace("\n", "").strip().split(',') for stack in traceback.format_stack()]
+    # Holds temporary callstack
+    # List with each element 4-tuple(filename, line number, function name, text)
+    # and filtered with respect to regular expressions
+    temp_call_stack = [frame for frame in traceback.extract_stack()
                        if not any(reg.match(frame[0]) for reg in REGULAR_EXPS)]
 
-    # avoid duplication.
-    if temp_call_stack not in STACK_BOOK[current_model] and TRACK_FLAG \
-            and not issubclass(current_model, tuple(HALT_TRACKING)):
-        STACK_BOOK[current_model].append(temp_call_stack)
-        log.info("logging new call stack for %s:\n %s", current_model, temp_call_stack)
+    final_call_stack = "".join(traceback.format_list(temp_call_stack))
+
+    def _should_get_logged(entity_name):
+        """ checks if current call stack of current entity should be logged or not
```

Comment:
All docstrings should read as regular English sentences. So capitalize the first letter and add ending period. Sure - it's a private method - but you should just follow the same convention everywhere.